op-e2e/check_scripts_test.go

```diff
@@ -0,0 +1,74 @@
+package op_e2e
+
+import (
+	"context"
+	"testing"
+	"time"
+
+	"github.com/ethereum/go-ethereum/common/hexutil"
+	"github.com/ethereum/go-ethereum/log"
+
+	"github.com/stretchr/testify/require"
+
+	fjordChecks "github.com/ethereum-optimism/optimism/op-chain-ops/cmd/check-fjord/checks"
+	"github.com/ethereum-optimism/optimism/op-service/testlog"
+)
+
+// TestCheckFjordScript ensures the op-chain-ops/cmd/check-fjord script runs successfully
+// against a test chain with the fjord hardfork enabled
+func TestCheckFjordScript(t *testing.T) {
+	InitParallel(t)
+	log := testlog.Logger(t, log.LevelInfo)
+
+	cfg := DefaultSystemConfig(t)
+	genesisActivation := hexutil.Uint64(0)
+	cfg.DeployConfig.L1CancunTimeOffset = &genesisActivation
+	cfg.DeployConfig.L2GenesisRegolithTimeOffset = &genesisActivation
+	cfg.DeployConfig.L2GenesisCanyonTimeOffset = &genesisActivation
+	cfg.DeployConfig.L2GenesisDeltaTimeOffset = &genesisActivation
+	cfg.DeployConfig.L2GenesisEcotoneTimeOffset = &genesisActivation
+
+	one := hexutil.Uint64(1)
+
+	tests := []struct {
+		name            string
+		fjordActivation *hexutil.Uint64
+		expectErr       bool
+	}{
+		{
+			name:            "fjord_activated",
+			fjordActivation: &one,
+			expectErr:       false,
+		},
+		{
+			name:            "fjord_unactivated",
+			fjordActivation: nil,
+			expectErr:       true,
+		},
```

Comment:
This is actually a very good idea, to test that the checks error without fjord activation. Even better would be to assert that every individual check, instead of the aggregate checks, errors. So we'd assert that all of these error individually
* checks.CheckRIP7212
* checks.CheckGasPriceOracle
* checks.CheckTxEmpty
* checks.CheckTxAllZero
* checks.CheckTxAll42
* checks.CheckTxRandom

but not the `CheckAll..`s

For the other test that the check script doesn't error on activated Fjord, it makes sense to just test the single `CheckAll` aggregate check.

Reply:
Done in https://github.com/ethereum-optimism/optimism/pull/10578/commits/23e875defd3c4e8c5d81d8f19f8eb8def430aa3f